roofit/roofitcore/src/FitHelpers.cxx

```diff
@@ -83,18 +96,47 @@ int calcAsymptoticCorrectedCovariance(RooAbsReal &pdf, RooMinimizer &minimizer,
          num(k, l) = 0.0;
       }
    }
-   RooArgSet obs;
-   pdf.getObservables(data.get(), obs);
+
    // Create derivative objects
    std::vector<std::unique_ptr<RooDerivative>> derivatives;
    const RooArgList &floated = rw->floatParsFinal();
    std::unique_ptr<RooArgSet> floatingparams{
-      static_cast<RooArgSet *>(pdf.getParameters(data)->selectByAttrib("Constant", false))};
+     static_cast<RooArgSet *>(logpdf.getParameters(data)->selectByAttrib("Constant", false))};
+
+   // Calculate derivatives of logpdf
    for (const auto paramresult : floated) {
       auto paraminternal = static_cast<RooRealVar *>(floatingparams->find(*paramresult));
       assert(floatingparams->find(*paramresult)->IsA() == RooRealVar::Class());
-      derivatives.emplace_back(pdf.derivative(*paraminternal, obs, 1));
+      double eps = 1.0e-4;
+      double error = static_cast<RooRealVar *>(paramresult)->getError();
+      //If parameter has range, undo multplication with range
+      if (paraminternal->hasMin() && paraminternal->hasMax())
+	derivatives.emplace_back(logpdf.derivative(*paraminternal, obs, 1, eps*error/(paraminternal->getMax()-paraminternal->getMin())));
+      else
+	derivatives.emplace_back(logpdf.derivative(*paraminternal, obs, 1, eps*error));	
    }
+   
+   // Calculate derivatives for number of expected events, needed for extended ML fit
+   RooAbsPdf *extended_pdf = dynamic_cast<RooAbsPdf*>(&pdf);
+   std::vector<double> diffs_expected(floated.getSize(), 0.0);
```

Comment:
```suggestion
   std::vector<double> diffs_expected(floated.size(), 0.0);
```
Please always use the standard-library-style `getSize()`, also for RooFit collections.

Reply:
Fine for me to use size() instead, thanks (I assume from your suggestion this is what you mean).